src/components/Table/utils/TableState.ts

```diff
@@ -105,7 +105,8 @@ export class TableState {
   }
 
   loadSelected(rows: GridDataRow<any>[]): void {
-    const selectedRows = rows.filter((row) => row.initSelected);
+    const childRows = rows.flatMap((r) => (r.children ? flattenRows(r.children) : []));
```

Comment:
I think this could be just:

```
const allRows = flattenRows(rows);
```

I.e. you can pass `rows` directly to `flattenRows` instead of re-coding some of `flattenRows` here.